### ISSUE TITLE
1.7.1 RPM warns "post remove script called with unknown argument"

### ISSUE THREAD

**Issue by loren** [elastic/elasticsearch#12606]
From https://discuss.elastic.co/t/1-7-1-rpm-warns-post-remove-script-called-with-unknown-argument/26693:

```
$ ESVERSION=1.7.1
$ package=elasticsearch-$ESVERSION.noarch.rpm && wget https://download.elasticsearch.org/elasticsearch/elasticsearch/$package -O /tmp/$package && sudo rpm -Uvh /tmp/$package
--2015-08-02 20:27:24--  https://download.elasticsearch.org/elasticsearch/elasticsearch/elasticsearch-1.7.1.noarch.rpm
Resolving download.elasticsearch.org... 174.129.211.252, 174.129.224.133, 23.21.179.119, ...
Connecting to download.elasticsearch.org|174.129.211.252|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 27323419 (26M) [application/x-redhat-package-manager]
Saving to: “/tmp/elasticsearch-1.7.1.noarch.rpm”

100%[===================================================>] 27,323,419  8.56M/s   in 3.0s

2015-08-02 20:27:27 (8.56 MB/s) - “/tmp/elasticsearch-1.7.1.noarch.rpm” saved [27323419/27323419]

warning: /tmp/elasticsearch-1.7.1.noarch.rpm: Header V4 RSA/SHA1 Signature, key ID d88e42b4: NOKEY
Preparing...                ########################################### [100%]
   1:elasticsearch          ########################################### [100%]
post remove script called with unknown argument `1'
warning: %postun(elasticsearch-1.7.0-1.noarch) scriptlet failed, exit status 1
```

Reported on Centos 6 and RHEL 6.6.

**Comment by spinscale** [comment]
Currently digging into this. This happens when you upgrade and the installed elasticsearch version is > 1.6...

**Comment by tlrx** [comment]
I spotted it and resolved it in #12298

**Comment by spinscale** [comment]
just found it as well. IMO the postrm fix should be backported into 1.7

**Comment by spinscale** [comment]
Closed by https://github.com/elastic/elasticsearch/commit/a370e2d2261c7e08e4b6d4f29cf7aacf4d8f5414

**Comment by tegansnyder** [comment]
I'm still encountering this issue. It doesn't look like the rpm has been updated on your website.

I quick peak at `elasticsearch-1.7.1.noarch.rpm` reveals:
```
    # RedHat ####################################################
    1)
        # If $1=1 this is an install
        IS_UPGRADE=false
    ;;
    2)
        # If $1=1 this is an upgrade
        IS_UPGRADE=true
    ;;
```

Per the commit @a370e2d2261c7e08e4b6d4f29cf7aacf4d8f5414 it should read more like:
```
    1)
        # If $1=1 this is an upgrade
        IS_UPGRADE=true
    ;;
```

**Comment by loren** [comment]
I think this fix is getting rolled into 1.7.2.

**Comment by tegansnyder** [comment]
@loren good to know. Thanks.
